odoo/addons/base/models/res_users.py

```diff
@@ -1753,7 +1753,8 @@ class APIKeyDescription(models.TransientModel):
     @check_identity
     def make_key(self):
         # only create keys for users who can delete their keys
-        if not self.user_has_groups('base.group_user'):
+        if not self.user_has_groups('base.group_user') and \
+           not tools.str2bool(self.env['ir.config_parameter'].sudo().get_param('portal.allow_api_keys')):
```

Comment:
Maybe just `get_param`? That seems to be what many of the other settings do, and `set_param(key, False)` removes it if it exists.